metpy/plots/skewt.py

```diff
@@ -770,6 +770,52 @@ def plot(self, u, v, **kwargs):
         u, v = delete_masked_points(u, v)
         return self.ax.plot(u, v, **line_args)
 
+    def plot_layer(self, u, v, p, hgt, bounds, colors, **kwargs):
+        r"""Plot different layers in different colors.
+
+        Plots the wind data on the hodograph colored
+        differently for different height layers. Designed to
+        accept layer bounds in meters AGL, to correspond with
+        storm relative helicity calculations.
+
+        Parameters
+        ----------
+        u : array_like
+            u-component of wind
+        v : array_like
+            v-component of wind
+        p : array_like
+            pressure in hPa
+        hgt: array_like
+            heights from sounding
+        bounds: array_like
+            array of layer bounds, starting
+            at the bottom of the first layer
+            and ending at the top of the last.
+        colors: array_like
+            array of strings containing the
+            colors for each layer of the
+            hodograph
+        kwargs
+            Other keyword arguments to pass to :meth:`matplotlib.axes.Axes.plot`
+
+        Returns
+        -------
+        list[matplotlib.lines.Line2D]
+            lines plotted
+
+        See Also
+        --------
+        :meth:`Hodograph.plot_colormapped`
+
+        """
+        line_args = self._form_line_args(kwargs)
+        u, v = delete_masked_points(u, v)
```

Comment:
We can condense this and ditch the loop :) I think we should make `plot_colormapped` polymorphic. You can accept either a color map for continuous coloring *or* accept a `bounds` and `colors` kwarg to do a segmented coloring. Here's the snippet:

```python
import matplotlib as mpl
#  Get data, etc.
cmap = mpl.colors.ListedColormap(['r', 'g', 'b'])
norm = mpl.colors.BoundaryNorm([0, 1, 3, 5], cmap.N)
h = Hodograph(component_range=80.)
h.plot_colormapped(u, v, hgt, cmap=cmap)
h.add_grid(increment=20
```
![unknown](https://user-images.githubusercontent.com/325269/28241889-8cbf42b6-6963-11e7-99ff-27149f739182.png)